File: app/src/main/java/com/example/simplehabitassignment/data/vos/models/TopicsModel.java
package com.example.simplehabitassignment.data.vos.models;

import com.example.simplehabitassignment.data.vos.TopicsVO;

import java.util.List;

public interface TopicsModel {

    void getTopics (String accessToken, int page, TopicsDelegate topicsDelegate);


    interface TopicsDelegate
    {
        void onSuccess(List<TopicsVO> topicsVO);
        void onFail(String errorMessage);

    }




}

File: app/src/main/java/com/example/simplehabitassignment/data/vos/models/TopicsModel.kt
package com.example.simplehabitassignment.data.vos.models

import com.example.simplehabitassignment.data.vos.TopicsVO

interface TopicsModel {

    fun getTopics(accessToken: String, page: Int, topicsDelegate: TopicsDelegate)


    interface TopicsDelegate {
        fun onSuccess(topicsVO: List<TopicsVO>)
        fun onFail(errorMessage: String)

    }


}
